test_conformance/images/kernel_read_write/test_common.cpp

```diff
@@ -34,20 +34,45 @@ cl_sampler create_sampler(cl_context context, image_sampler_data *sdata, bool te
     return sampler;
 }
 
+void get_image_dimensions(image_descriptor *imageInfo, size_t &width,
+                          size_t &height, size_t &depth)
+{
+    width = imageInfo->width;
+    height = 1;
+    depth = 1;
+    switch (imageInfo->type)
+    {
+        case CL_MEM_OBJECT_IMAGE1D: break;
+        case CL_MEM_OBJECT_IMAGE1D_ARRAY: height = imageInfo->arraySize; break;
+        case CL_MEM_OBJECT_IMAGE2D: height = imageInfo->height; break;
+        case CL_MEM_OBJECT_IMAGE2D_ARRAY:
+            height = imageInfo->height;
+            depth = imageInfo->arraySize;
+            break;
+        case CL_MEM_OBJECT_IMAGE3D:
+            height = imageInfo->height;
+            depth = imageInfo->depth;
+            break;
+    }
```

Comment:
The _errCode_ argument to _test_error_ should only ever be an OpenCL error code.

Using `test_error(-1, "ERROR: <some message>")` will print:

```
ERROR: ERROR: <some message> (CL_DEVICE_NOT_FOUND from filename:lineno)
```

It will also return _CL_DEVICE_NOT_FOUND_.  Could you use _log_error_ and have this return _TEST_FAIL_ instead, please?  This change will need to propagate up a bit, to _InitFloatCoordsCommon_ and to the call to it in _test_read_image_.

Reply:
The `test_error` issue has been dealt with.  @jlewis-austin, does the change here look good to you too?

Reply:
yes, thanks!